### ISSUE TITLE
Support recurring jobs

### ISSUE THREAD

**Issue by alfpark** [Azure/batch-shipyard#19]
Support `JobSchedule`s and recurrences. See issue #15 for more details.

- [x] Json schema for job schedule/recurrence
- [x] Generic job manager
  - [x] Docker image
  - [x] Spec transfer
  - [x] Spec reader/submitter
  - [x] Optional task monitor
- [x] `jobs list` should include both regular jobs and schedules
- [x] `jobs del`/`jobs term`/`jobs disable`/`jobs enable`/`jobs migrate` should detect schedules
  - [x] `--jobscheduleid` flag to explicitly act on schedule rather than job
  - [x] disambiguate if `--all` flag exists
- [x] Update configuration doc
- [x] Update usage doc

**Comment by lediur** [comment]
Would it be worth it for me to try contributing support for JobSchedules for single jobs out of band for 3.0? It'd be worth it for me to be able to reliably schedule Docker containers as recurring jobs, even if I can only run one task per job.

**Comment by alfpark** [comment]
@lediur Unfortunately, this feature is scoped for multiple tasks which requires quite a bit of internal change. I'd like to keep the code path the same so it's more readily maintainable, regardless of the number of tasks submitted for recurrence. That being said, I might be able to do this out-of-band from 3.0.0 (since that release is quite a bit away). How big of a blocker is this for you?

**Comment by lediur** [comment]
It's not a big blocker - I was actually thinking about using a CI tool like VSTS Builds to schedule recurring batch-shipyard jobs as a workaround until JobSchedules are implemented.

**Comment by lediur** [comment]
Can we add auto-pools under this category? That would be a fantastic addition.

Seems like it would entail adding a property to scheduled job specifications that specify a pool description in `pool.json` to pair up with.

It also seems like the "native" auto-pools are only for scheduled jobs. Something like a `jobs add --wait --autostart`, where shipyard would start the pool if it doesn't exist, start the job, wait for job completion, and then clean everything up, would be great to have. Does the Azure Batch SDK have the capability to wait on job completion, or do you have to just repeatedly query for status?

**Comment by alfpark** [comment]
Autopool support is possible (even for non recurring jobs). I'll split this to another issue.
